portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -9,24 +9,122 @@
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+// See the License for the specific language gover
+
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 import java.io.IOException;
+import com.google.sps.data.Task;
+import java.util.Arrays;
+import com.google.sps.data.ServerStats;
+import com.google.gson.Gson;
+import java.util.Date;
+import java.io.PrintWriter;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
-
+public final class DataServlet extends HttpServlet {
+    private final Date startTime = new Date();
   @Override
+  
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+   // response.setContentType("text/html;");
+   // response.getWriter().println("Hello Chenxi!");
+
+
+
+
+    ///*
+    Date currentTime = new Date();
+    long maxMemory = Runtime.getRuntime().maxMemory();
+    long usedMemory = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
+
+    // Convert the server stats to JSON
+    ServerStats serverStats = new ServerStats(startTime, currentTime, maxMemory, usedMemory);
+    String json = convertToJson(serverStats);
+    // Send the JSON as the response
```

Comment:
usually a comment is not necessary if the code logic and variable name is obvious enough, although it's up to you to decide whether you should add comment